src/yamerl_node_map.erl

```diff
@@ -173,6 +173,20 @@ set_kv(#yamerl_constr{detailed_constr = false},
     Data = maps:put(Key, Value, Map),
     Builder#map_builder{state = none, data = Data};
 
+set_kv(#yamerl_constr{detailed_constr = true, allow_duplicate_keys = true},
```

Comment:
Here, you can look at the content of `ext_options` to find the option you added. That said, you will have to do that in the function body, not from a pattern match in the function arguments. A solution is to extract the option before calling `set_kv()` and passing the flag to `set_kv()`.

Reply:
I went with the variant in de body. Only thing I dislike is that we now have a nested case statement.